Internal Server error

1. Warning - during compiling the code
g++ -c -o main.o src/main.cpp 
g++ -c -o udpTracker.o src/udpTracker.cpp 
src/udpTracker.cpp: In static member function ‘static int UDPT::UDPTracker::resolveRequest(UDPT::UDPTracker*, SOCKADDR_IN*, char*, int)’:
src/udpTracker.cpp:524:59: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   cout << ":: " << (void*)m_hton32(remote->sin_addr.s_addr) << ": " << m_hton16(remote->sin_port) << " ACTION=" << action << endl;
                                                           ^
g++ -c -o database.o src/db/database.cpp 
g++ -c -o driver_sqlite.o src/db/driver_sqlite.cpp 
g++ -c -o settings.o src/settings.cpp 
cc -c -o tools.o src/tools.c 
g++ -c -o httpserver.o src/http/httpserver.cpp 
g++ -c -o webapp.o src/http/webapp.cpp 
Linking...
g++  -O3 -o udpt main.o udpTracker.o database.o driver_sqlite.o settings.o tools.o httpserver.o webapp.o -lsqlite3 -lpthread
Done.

2. config file i don't know if it is correct.
; udpt Settings File - Created Automatically.
[apiserver]
enable=1
port=6969
threads=1

[database]
driver=sqlite3
file=tracker.db

[tracker]
allow_iana_ips=1
allow_remotes=1
announce_interval=1800
cleanup_interval=120
is_dynamic=0
local_subnet=10.10.0   <my pc ip is 10.10.0.6> and <my router is 10.10.0.1> so i believe its correct
port=6969
remote_ip=112.12.14.117 - from what is my ip google search / my external ip
threads=5

i can access 127.0.0.1:6969 tracker is running
i can also access 112.12.14.117:6969 tracker is running after port forwarding
firewall is configured properly on my linux mint 19.

3. REAL PROBLEM 
i created a private torrent and tried to add the info hash using browser url 
http://127.0.0.1:6969/api?action=add&hash=66d65393825343a65b073c5a862a8ad869d8cc54
I cannot add info hash through api it says 
Internal Server Error
An Error Occurred while trying to process your request.

I tried creating an db entry using sqlite DB browser still no luck.

Please help....

--- a/README.md
+++ b/README.md
@@ -49,6 +49,10 @@ $ ./udpt -d [udpt.log] [udpt.conf]
 Cleaning:
 $ make clean
 
+Adding torrent using HTTP API:
+https://127.0.0.1?auth=admin&action=add&hash=TORRENT_HASH
+you can change the ip and auth value in udpt.conf file under [api.keys]
+
 This software currently uses the Sqlite3 Library (public domain).
 
 Developed by Naim A. <naim94a@gmail.com>.

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -144,6 +144,7 @@ int main(int argc, char *argv[])
 		const char strDATABASE[] = "database";
 		const char strTRACKER[] = "tracker";
 		const char strAPISRV [] = "apiserver";
+		const char strAPIKEY [] = "api.keys";
 		// set default settings:
 
 		settings->set (strDATABASE, "driver", "sqlite3");
@@ -162,6 +163,9 @@ int main(int argc, char *argv[])
 		settings->set (strAPISRV, "enable", "1");
 		settings->set (strAPISRV, "threads", "1");
 		settings->set (strAPISRV, "port", "6969");	// TCP PORT
+
+		settings->set(strAPIKEY, "admin", "127.0.0.1");
+
 
 		settings->save();
 		cerr << "Failed to read from '" << config_file.c_str() << "'. Using default settings." << endl;

